nesta/packages/nlp_utils/embed_clustering.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from sklearn.mixture import GaussianMixture
+
+np.random.seed(42)
+
+
+def vals2arr(d):
+    """Store dictionary keys and values in arrays."""
+    return list(d.keys()), np.array(list(d.values()))
+
+
+def clustering(vectors):
+    """Vector clustering with Gaussian Mixtures.
+
+    Args:
+        vectors (:obj:`numpy.array` of :obj:`float`): Document vectors.
+
+    Return:
+        gmm (:obj:`sklearn.mixture.gaussian_mixture.GaussianMixture`): Fitted Gaussian Mixture model.
+
+    """
+    # For now, GMMs hyperparameters are hardcoded.
+    gmm = GaussianMixture(covariance_type='tied', init_params='kmeans', max_iter=100,
+                          means_init=None, n_components=80, n_init=1, precisions_init=None,
+                          random_state=42, reg_covar=1e-06, tol=0.001, verbose=0,
+                          verbose_interval=10, warm_start=False, weights_init=None)
+    gmm.fit(vectors)
+    return gmm
+
+
+def filter_arr(arr, thresh=.2):
```

Comment:
this might be better in a general purpose `np_utils.py` in a function titled `arr2dict` with an optional arg `thresh` like so:

```
def arr2dic(arr, thresh=None):
    if thresh is not None:
        ids = np.where(arr <= thresh)[0]
    else:
        # get all the ids
     # return the ids
```

Reply:
Created a new `np_utils.py` file under the `misc_utils`